src/Notifier.cpp

```diff
@@ -163,9 +183,9 @@ size_t Notifier::ReceiveConfirmation() {
  * Reset the Shotgun-Alien queue to nullptr
  */
 void Notifier::Reset() {
-   std::lock_guard<std::mutex> guard(gLock);
    gQueue.reset(nullptr);
    gHandshakeQueue.reset(nullptr);
+   mKeepRunning.store(false);
```

Comment:
Also. If this is to be called from two threads, which is really the only reason to use an atomic, then you have a **bad race** here where the background thread might be using the internals while the calling thread is forcing the same internals to be destroyed.